pgcli/pgcompleter.py

```diff
@@ -385,15 +392,27 @@ def get_column_matches(self, suggestion, word_before_cursor):
 
         return self.find_matches(word_before_cursor, flat_cols, meta='column')
 
-    def generate_alias(self, tbl, tbls):
+    def alias(self, tbl, tbls):
         """ Generate a unique table alias
         tbl - name of the table to alias, quoted if it needs to be
-        tbls - set of table refs already in use, normalized with normalize_ref
+        tbls - TableReference iterable of tables already in query
         """
+        tbl = self.case(tbl)
+        tbls = set(normalize_ref(t.ref) for t in tbls)
+        if self.generate_aliases:
```

Comment:
Let's break off the specific chunk of code that transforms a table name into an alias into a standalone function. That way 1) it's easy to test just the alias generation, and 2) savvy users will be able to monkey patch in their own custom alias generation by replacing just that function